20_classes_lab/03.py

```diff
@@ -1,13 +1,52 @@
-"""
-Following code assumes a class named Widget that depends on
-other widgets
-Widget ctor takes a name, and function 
-add_dependency(depends)
-marks widget as depends on all widgets in "depends" list
-
-When building a widget, you need to first build all the widgets it
-depends on.
-
-Implement widget so the following code works
-"""
-
+#!/usr/bin/env python
+
+__author__ = 'Eabel'
+
+import sys
+import os
+
+
+class Widget(object):
+    
+
+    def __init__(self,name):
+        self._name = name
+        self._dependencies = set()
+
+
+    def add_dependency(self,*modules):
+        for module in modules:
+            self._dependencies.add(module._name)
+            self._dependencies.update(module._dependencies)
+            
+
+    def build(self):
+        print ', '.join(self._dependencies)
+

```

Comment:
I don't think this works as requested.

Dependencies should be printed in an order such that no module is printed before all of its dependencies are printed.